edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/RouteTest.cs

```diff
@@ -19,29 +20,29 @@ public class RouteTest : RoutingUnitTestBase
         static readonly Endpoint Endpoint2 = new TestEndpoint("id2");
         static readonly Endpoint Endpoint3 = new TestEndpoint("id1");
 
-        static readonly Route Route1 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint1 });
-        static readonly Route Route2 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint2 });
-        static readonly Route Route3 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint3 });
-        static readonly Route Route4 = new Route("id2", "rule2", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint1 });
-        static readonly Route Route5 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>());
-        static readonly Route Route6 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>());
-        static readonly Route Route7 = new Route("id2", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint1 });
+        static readonly Route Route1 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
+        static readonly Route Route2 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint2, 0, 0);
+        static readonly Route Route3 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint3, 0, 0);
+        static readonly Route Route4 = new Route("id2", "rule2", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
+        static readonly Route Route5 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0);
```

Comment:
why test data is changed? previously it is empty endpoint, and now a mock endpoint is provided.

Reply:
the class paradigm changed, previous it took a collection, which could be empty.  That's a bug and not valid for Edge scenarios.  For our routes, there must be one endpoint and one endpoint only, now the code is updated to reflect that.

---
In reply to: [377461524](https://github.com/Azure/iotedge/pull/2469#discussion_r377461524) [](ancestors = 377461524)